memory-bank/01-project-overview.md

```diff
@@ -0,0 +1,93 @@
+# Apache Traffic Server - Project Overview
+
+## Project Overview
+
+**Apache Traffic Server (ATS)** is a high-performance, open-source HTTP/HTTPS caching proxy server and web application acceleration platform. It's designed as a building block for cloud services and large-scale web applications.
+
+### Key Characteristics
+- **Version**: 10.2.0 (current)
+- **Language**: Primarily C++ (C++20 standard)
+- **Build System**: CMake
+- **License**: Apache License 2.0
+- **Platform Support**: Linux, FreeBSD, macOS, with extensive CI across multiple distributions
+
+## Architecture Overview
+
+### Core Components
+
+#### 1. **Traffic Server Core** (`src/traffic_server/`)
+- Main proxy server executable
+- Event-driven architecture
+- Multi-threaded processing
+
+#### 2. **I/O Core** (`src/iocore/`)
+- **AIO**: Asynchronous I/O operations
+- **Cache**: Disk and RAM caching system
+- **DNS**: Asynchronous DNS resolution
+- **Event System**: Event-driven engine foundation
+- **HostDB**: Internal DNS cache
+- **Net**: Network layer with QUIC support
+- **Utils**: Core utilities
+
+#### 3. **Proxy Logic** (`src/proxy/`)
+- **HTTP**: HTTP/1.1 protocol implementation (`http/`)
+- **HTTP/2**: HTTP/2 support (`http2/`)
+- **HTTP/3**: HTTP/3 implementation (`http3/`)
+- **Headers**: Header parsing and management (`hdrs/`)
+- **Logging**: Flexible logging system (`logging/`)
+- **Shared**: Common proxy components (`shared/`)
+
+#### 4. **Management** (`src/mgmt/`)
+- JSONRPC server for management
+- Configuration management
+- Administrative tools
```

Comment:
Should the directories be on all of these for each section?  Maybe just leave that for the README.md?